pkgs/test/by-name/src/references.rs

```diff
@@ -0,0 +1,185 @@
+                    "{}: Path {} is a symlink which cannot be resolved: {e}.",
+                    context.relative_package_dir.display(),
+                    subpath.display(),
+                ))?;
+            }
+        }
+    } else if path.is_dir() {
+        // Recursively check each entry
+        for entry in utils::read_dir_sorted(&path)? {
+            let entry_subpath = subpath.join(entry.file_name());
+            check_path(context, &entry_subpath)
+                .context(format!("Error while recursing into {}", subpath.display()))?
+        }
+    } else if path.is_file() {
+        // Only check Nix files
+        if let Some(ext) = path.extension() {
+            if ext == OsStr::new("nix") {
+                check_nix_file(context, subpath).context(format!(
+                    "Error while checking Nix file {}",
+                    subpath.display()
+                ))?
+            }
+        }
+    } else {
+        // This should never happen, git doesn't support other file types
+        anyhow::bail!("Unsupported file type for path {}", subpath.display());
+    }
+    Ok(())
+}
+
+/// Check whether a nix file contains path expression references pointing outside the package
+/// directory
+fn check_nix_file<W: io::Write>(
+    context: &mut PackageContext<W>,
+    subpath: &Path,
+) -> anyhow::Result<()> {
+    let path = context.absolute_package_dir.join(subpath);
+    let parent_dir = path.parent().context(format!(
+        "Could not get parent of path {}",
+        subpath.display()
+    ))?;
+
+    let contents =
+        read_to_string(&path).context(format!("Could not read file {}", subpath.display()))?;
+
+    let root = Root::parse(&contents);
+    if let Some(error) = root.errors().first() {
+        context.writer.write(&format!(
+            "{}: File {} could not be parsed by rnix: {}",
+            context.relative_package_dir.display(),
+            subpath.display(),
+            error,
+        ))?;
+        return Ok(());
+    }
+
+    let line_index = LineIndex::new(&contents);
+
+    for node in root.syntax().descendants() {
+        // We're only interested in Path expressions
+        if node.kind() != NODE_PATH {
+            continue;
+        }
+
+        let text = node.text().to_string();
+        let line = line_index.line(node.text_range().start().into());
+
+        // Filters out ./foo/${bar}/baz
+        // TODO: We can just check ./foo
+        if node.children().count() != 0 {
+            context.writer.write(&format!(
+                "{}: File {} at line {line} contains the path expression \"{}\", which is not yet supported and may point outside the directory of that package.",
+                context.relative_package_dir.display(),
+                subpath.display(),
+                text
+            ))?;
+            continue;
+        }
+
+        // Filters out search paths like <nixpkgs>
+        if str::starts_with(&text, "<") {
```

Comment:
I'm pretty sure this is equivalent. Tests pass with this change ¯\\_(ツ)_/¯
```suggestion
        if text.starts_with('<') {
```